frontend/coprs_frontend/coprs/views/apiv3_ns/schema/schemas.py

```diff
@@ -0,0 +1,464 @@
+
+        return api.model(self.__class__.__name__, self.schema)
+
+
+@dataclass
+class PaginationMeta(Schema):
+    limit: Integer
+    offset: Integer
+    order: String
+    order_type: String
+
+
+_pagination_meta_model = PaginationMeta.get_cls().model()
+
+
+# checks if `items` in Pagination schema are defined (don't have None value)
+def _check_if_items_are_defined(method):
+    @wraps(method)
+    def check_items(self, *args, **kwargs):
+        if getattr(self, "items") is None:
+            raise KeyError(
+                "No items are defined in Pagination. Perhaps you forgot to"
+                " specify it when creating Pagination instance?"
+            )
+        return method(self, *args, **kwargs)
+
+    return check_items
+
+
+@dataclass
+class Pagination(Schema):
+    """
+    Pagination items can be basically anything (any schema) so specify a model to
+     `items` like this: build_pagination_model = Pagination(items=build_model).model()
+    """
+
+    items: Any = None
+    meta: Nested = Nested(_pagination_meta_model)
+
+    @_check_if_items_are_defined
+    def model(self):
+        return super().model()
+
+
+@dataclass
+class _ProjectChrootFields:
+    additional_repos: List
+    additional_packages: List
+    additional_modules: List
+    with_opts: List
+    without_opts: List
+    isolation: String
+
+
+@dataclass
+class ProjectChroot(_ProjectChrootFields, Schema):
+    mock_chroot: String
+    ownername: String
+    projectname: String
+    comps_name: String
+    delete_after_days: Integer
+
+
+@dataclass
+class ProjectChrootGet(InputSchema):
+    ownername: String
+    projectname: String
+    chrootname: String = mock_chroot
+
+    __all_required: bool = True
+
+
+@dataclass
+class Repo(Schema):
+    baseurl: Url
+    module_hotfixes: Boolean
+    priority: Integer
+    id_field: String = String(example="copr_base")
+    name: String = String(example="Copr repository")
+
+
+_repo_model = Repo.get_cls().model()
+
+
+@dataclass
+class ProjectChrootBuildConfig(_ProjectChrootFields, Schema):
+    chroot: String
+    enable_net: Boolean
+    repos: List = List(Nested(_repo_model))
+
+
+@dataclass
+class _SourceDictScmFields:
+    clone_url: String
+    committish: String
+    spec: String
+    subdirectory: String
+
+
+@dataclass
+class SourceDictScm(_SourceDictScmFields, Schema):
+    source_build_method: String
+    type: String = scm_type
+
+
+@dataclass
+class SourceDictPyPI(Schema):
+    pypi_package_name: String
+    pypi_package_version: String
+    spec_generator: String
+    spec_template: String
+    python_versions: List
+
+
+@dataclass
+class SourcePackage(Schema):
+    name: String
+    url: String
+    version: String
+
+
+_source_package_model = SourcePackage.get_cls().model()
+
+
+@dataclass
+class Build(Schema):
+    chroots: List
+    ended_on: Integer
+    id_field: Integer
+    is_background: Boolean
+    ownername: String
+    project_dirname: String
+    projectname: String
```

Comment:
I am quite unhappy, that we define the same data type on multiple places. Here, and in `fields.py`.

```python
projectname = String(
    description="Name of the project",
    example="copr-dev",
)
```

Can we do something about that?